src/main/java/com/josdem/vetlog/model/Vaccination.java

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2024 Jose Morales contact@josdem.io
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+
+package com.josdem.vetlog.model;
+
+import static jakarta.persistence.EnumType.STRING;
+import static jakarta.persistence.GenerationType.IDENTITY;
+
+import com.josdem.vetlog.enums.PetType;
+import jakarta.persistence.*;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+
+@Entity
+@Getter
+@Setter
+@ToString
+public class Vaccination {
+
+    @Id
+    @GeneratedValue(strategy = IDENTITY)
+    private Long id;
+
+    @Column(nullable = false)
+    private String name;
+
+    @Column(nullable = false)
+    private LocalDate date;
+
+    @Enumerated(EnumType.STRING)
+    @Column(nullable = false)
+    private Status status;
+
```

Comment:
Please create a different file for this status as enum, something like: `VaccinationStatus`. The reason is that in future this will evolve to get a printable value for Thymeleaf, take a look at: https://github.com/josdem/vetlog-spring-boot/blob/main/src/main/java/com/josdem/vetlog/enums/PetType.java and usage here: https://github.com/josdem/vetlog-spring-boot/blob/main/src/main/resources/templates/pet/create.html

Reply:
Fixed